### ISSUE TITLE
Make "repo badges hiding" optional

### ISSUE THREAD

**Issue by tomkerkhove** [refined-github/refined-github#5268]
### Description

Provide capability to keep on showing the default GitHub repo visibility badge which is safer to always show instead of just an icon.

Having this as an opt-in is fine for me.

### Screenshot

![image](https://user-images.githubusercontent.com/4345663/148404335-3fcac428-1fbc-4d50-904f-f03bdab0dfb9.png)

### Example URL

N/A

**Comment by cheap-glitch** [comment]
I have no opinion about this, but since [several people asked about it already](https://github.com/refined-github/refined-github/pull/5125#issuecomment-982868997) and it's very easy to do, I say we make this feature switchable.

**Comment by fregante** [comment]
This is one of those features that are easily undone with some CSS in the options, which is what the field is there for 😃

**Comment by cheap-glitch** [comment]
> This is one of those features that are easily undone with some CSS in the options, which is what the field is there for

True, but having a proper checkbox for it would make it easier for the users, and would avoid further issues like this one, at very little cost. Not to mention that not everybody who uses RGH is familiar with CSS.

Another issue with custom CSS is that it has to keep up with GitHub's DOM updates. If we update the selectors used in the feature, the users who want to keep it disabled will have to update their custom CSS as well. So in a way, we needlessly push on them that (arguably small) burden of maintenance.

**Comment by tomkerkhove** [comment]
> This is one of those features that are easily undone with some CSS in the options, which is what the field is there for 😃

If you know how to write CSS 😅 Which is not the case for me, unfortunately.

> Not to mention that not everybody who uses RGH is familiar with CSS.

This is the case for me, sorry :)

**Comment by cob16** [comment]
With the latest version of the plugin, I see no badge for either private or public repos. Is there another way you are able to tell the difference in Github I am missing (when you do not have admin on the repo)?

My day to day work requires that I know what repos are public and what are private so sadly I have had to disable this plugin for now :(

**Comment by tomkerkhove** [comment]
Unfortunately I'm in the same boat where the only way to see it is different icons.

**Public:**
![image](https://user-images.githubusercontent.com/4345663/148543771-14c9964a-f455-4352-bf77-35d50b845e3b.png)

**Private:**
![image](https://user-images.githubusercontent.com/4345663/148543824-5a584c8a-3633-4dda-a055-d3ec8f0aa8d3.png)

Which are easily overlooked IMO and the reason why I'd love to be able to use the default GitHub way. I've even considered turning off GitHub Refined because I don't want to mess up which is my last resort that I'm holding off.

**Comment by kidonng** [comment]
Since this is brought up again, I want to make it clear that [I'm not a fan of hiding the labels completely](https://github.com/refined-github/refined-github/issues/4767#issuecomment-965182086):

> That's my first thought when GitHub  decided to show labels everywhere, the question being that dropping the label completely may lead to another extreme, as **the icon can not represent multiple repository types and may not make sense on its own for some people**.

And quoting [GitHub](https://github.blog/changelog/2021-10-11-all-repositories-are-labeled-public-private-or-internal/):

> Public repositories now have a `public` label next to their names like private and internal repositories do.
>
> Previously, repositories had a label next to their name that indicated if they were `private` or `internal`, but there was no label if they were `public`. **This led to people not recognizing that a repository was public and accidentally committing private code to it.** Now, public repositories have a `public` label next to their name like private and internal repositories do.

As of now 4 people asked for a way to disable this feature, does that count as ["too many people complain"](https://github.com/refined-github/refined-github/issues/4767#issuecomment-971596542)? If so, we should take actions.

> This is one of those features that are easily undone with some CSS in the options, which is what the field is there for 😃

The problem of this approach is that not everyone is seasoned enough to know how to revert the styles. I bet most people don't even know **what keywords should be used search for those styles**, even if we post them somewhere.

And it returns to the problem that [RGH have few user docs than it should](https://github.com/refined-github/refined-github/issues/4288#issuecomment-829353106) (yes, maintainers are users too). Too many details are just buried in the conversations and rediscovered from time to time. This is especially unfriendly for casual users.

**Comment by fregante** [comment]
Should we drop it? It was never worth the hassle for me

**Comment by tomkerkhove** [comment]
I'm fine with making it an opt-out/in; I don't want to force the badges on everyone but it's up to the maintainers of course.

Sorry for the hassle.

**Comment by jpeterson** [comment]
Love this project! Thanks for everything.

Just lending another perspective: my GH Org makes heavy use of [private and internal repos](https://docs.github.com/en/enterprise-cloud@latest/repositories/creating-and-managing-repositories/about-repositories#about-repository-visibility), which use the same icon (internal repos are sort of a key feature to support [InnerSource](https://innersourcecommons.org/)). Hiding the label makes this difference indistinguishable.

I was able to revert this via custom CSS... but I think enough users may need to distinguish private/internal repos that this should be easier to toggle.

Here's the CSS I used for anyone interested:

```css
:is(#repository-container-header h1, .pinned-item-list-item-content) .Label {
	display: inline-block !important;
}
```
